test/test_models.py

```diff
@@ -33,6 +39,103 @@ def get_available_video_models():
+    # We hardcode it here to avoid having to re-generate the reference files
+    expected_file = expected_file = os.path.join(expected_file_base, 'ModelTester.test_' + name)
+    expected_file += "_expect.pkl"
+
+    if not ACCEPT and not os.path.exists(expected_file):
+        raise RuntimeError(
+            f"No expect file exists for {os.path.basename(expected_file)} in {expected_file}; "
+            "to accept the current output, re-run the failing test after setting the EXPECTTEST_ACCEPT "
+            "env variable. For example: EXPECTTEST_ACCEPT=1 pytest test/test_models.py -k alexnet"
+        )
+
+    return expected_file
+
+
+def _assert_expected(output, name, prec):
+    """Test that a python value matches the recorded contents of a file
+    based on a "check" name. The value must be
+    pickable with `torch.save`. This file
+    is placed in the 'expect' directory in the same directory
+    as the test script. You can automatically update the recorded test
+    output using an EXPECTTEST_ACCEPT=1 env variable.
+    """
+    expected_file = _get_expected_file(name)
+
+    if ACCEPT:
+        filename = {os.path.basename(expected_file)}
+        print("Accepting updated output for {}:\n\n{}".format(filename, output))
+        torch.save(output, expected_file)
+        MAX_PICKLE_SIZE = 50 * 1000  # 50 KB
+        binary_size = os.path.getsize(expected_file)
+        if binary_size > MAX_PICKLE_SIZE:
+            raise RuntimeError("The output for {}, is larger than 50kb".format(filename))
+    else:
+        expected = torch.load(expected_file)
+        rtol = atol = prec
+        torch.testing.assert_close(output, expected, rtol=rtol, atol=atol, check_dtype=False)
+
+
+def _check_jit_scriptable(nn_module, args, unwrapper=None, skip=False):
+    """Check that a nn.Module's results in TorchScript match eager and that it can be exported"""
+
+    def assert_export_import_module(m, args):
+        """Check that the results of a model are the same after saving and loading"""
+        def get_export_import_copy(m):
+            """Save and load a TorchScript model"""
+            buffer = io.BytesIO()
+            torch.jit.save(m, buffer)
+            buffer.seek(0)
+            imported = torch.jit.load(buffer)
+            return imported
+
+        m_import = get_export_import_copy(m)
+        with freeze_rng_state():
+            results = m(*args)
+        with freeze_rng_state():
+            results_from_imported = m_import(*args)
+        tol = 3e-4
+        try:
+            assert_equal(results, results_from_imported, atol=tol, rtol=tol)
+        except pytest.UsageError:
+            # custom check for the models that return named tuples:
+            # we compare field by field while ignoring None as assert_equal can't handle None
+            for a, b in zip(results, results_from_imported):
+                if a is not None:
+                    assert_equal(a, b, atol=tol, rtol=tol)
```

Comment:
This is the only bit that I had to modify: The previous call to `self.assertEqual` was able to check for nested structures with None inside, but `assert_close` cannot do it

(unrelated: I'll switch to `assert_close` instead of `assert_equal` before merging)